Benchmark: use `-Dconfig=production` over development

kprotty opened the pull request:
The benchmark is meant to asses performance in a likely production scenario. The default (development) config has `process.verify=true` which helps for double checking assertions but slows down normal operations. Using the production config in benchmark disables `verify`.

## Pre-merge checklist

Performance:

* [x] Compare `zig benchmark` on linux before and after this pr.
    ``` sh
    # benchmark results before
    1222 batches in 171.78 s
    load offered = 1000000 tx/s
    load accepted = 58215 tx/s
    batch latency p00 = 17 ms
    batch latency p10 = 34 ms
    batch latency p20 = 36 ms
    batch latency p30 = 37 ms
    batch latency p40 = 39 ms
    batch latency p50 = 41 ms
    batch latency p60 = 43 ms
    batch latency p70 = 55 ms
    batch latency p80 = 66 ms
    batch latency p90 = 81 ms
    batch latency p100 = 8822 ms
    transfer latency p00 = 17 ms
    transfer latency p10 = 4195 ms
    transfer latency p20 = 12929 ms
    transfer latency p30 = 23362 ms
    transfer latency p40 = 36108 ms
    transfer latency p50 = 52999 ms
    transfer latency p60 = 74178 ms
    transfer latency p70 = 95342 ms
    transfer latency p80 = 118262 ms
    transfer latency p90 = 142472 ms
    transfer latency p100 = 161747 ms
    
    # benchmark results after
    1222 batches in 167.85 s
    load offered = 1000000 tx/s
    load accepted = 59575 tx/s
    batch latency p00 = 16 ms
    batch latency p10 = 34 ms
    batch latency p20 = 35 ms
    batch latency p30 = 37 ms
    batch latency p40 = 38 ms
    batch latency p50 = 40 ms
    batch latency p60 = 46 ms
    batch latency p70 = 58 ms
    batch latency p80 = 67 ms
    batch latency p90 = 81 ms
    batch latency p100 = 8663 ms
    transfer latency p00 = 16 ms
    transfer latency p10 = 3975 ms
    transfer latency p20 = 12631 ms
    transfer latency p30 = 22563 ms
    transfer latency p40 = 34791 ms
    transfer latency p50 = 51093 ms
    transfer latency p60 = 71518 ms
    transfer latency p70 = 92318 ms
    transfer latency p80 = 115574 ms
    transfer latency p90 = 138775 ms
    transfer latency p100 = 157821 ms
    ```

jamii commented:
Could we instead default to production config in the benchmarks?

Having verify on by default has caught bugs that we didn't know to look for.

kprotty commented:
Curious about how `direct_io_required=true` in `default_production` will affect macos, but that sounds like a better idea.

jorangreef commented:
Let's do that @jamii and @kprotty.

Commit:
benchmark: use production config to disable verify

`constats.verify=true` enables some extra checks that
slow down normal operation and are only needed when
bug hunting or introducing new features. Disabling them
during normal development restricts the ability to catch
accidental bugs that go unnoticed.

Instead, have the benchmark replica explicitly run in
`-Dconfig=production` which disables `verify` while currently
having similar runtime values to the development config.

Commit:
benchmark: use production config for benchmark.zig

kprotty commented:
Switched over to specifically the benchmark. Seems like using `-Dconfig=production` improves results more on `benchmark.zig` than `main.zig`

jamii commented:
I'm surprised that has so little effect. Just another reminder that I have no idea what the actual bottlenecks are :S

jorangreef commented:
> I'm surprised that has so little effect. Just another reminder that I have no idea what the actual bottlenecks are :S

@jamii It's going to be so much fun when we find them...
